src/UI/Implementation/Component/Triggerer.php

```diff
@@ -65,12 +65,28 @@ protected function setTriggeredSignal(Component\Signal $signal, $event) {
 	}
 
 	/**
-	 * @return \ILIAS\UI\Implementation\Component\TriggeredSignalInterface[]
+	 * @return \ILIAS\UI\Implementation\Component\TriggeredSignal[]
 	 */
 	public function getTriggeredSignals() {
 		return $this->flattenArray($this->triggered_signals);
 	}
 
+	/**
+	 * Get signals that are triggered for a certain event.
+	 *
+	 * @param	string
+	 * @return \ILIAS\UI\Component\Signal
+	 */
+	public function getTriggeredSignalsFor($event) {
+		if (!isset($this->triggered_signals[$event])) {
+			return [];
+		}
+		return array_map(
+			function($ts) { return $ts->getSignal(); },
+			$this->triggered_signals["click"]
```

Comment:
This should be `$this->triggered_signals[$event]` as well, right?

Reply:
Yes. And the test should find it. Good catch, thx!